californium-core/src/main/java/org/eclipse/californium/core/network/serialization/DataSerializer.java

```diff
@@ -93,7 +93,7 @@ public final RawData serializeRequest(final Request request, final MessageCallba
 		byte[] body = serializeOptionsAndPayload(request);
 
 		MessageHeader header = new MessageHeader(CoAP.VERSION, request.getType(), request.getToken(),
-				request.getCode().value, request.getMID(), body.length);
+				request.getRawCode(), request.getMID(), body.length);
```

Comment:
This is a great change :-) IMHO it would now be sufficient to change `Request.getRawCode()` to return 0 if no `CoAP.Code` is set ... we wouldn't need to add the `EMPTY` value to the enum ...

WDYT?